pkg/generate/app/cmd/newapp.go

```diff
@@ -1113,6 +1112,25 @@ func (c *AppConfig) run(acceptors app.Acceptors) (*AppResult, error) {
 	}, nil
 }
 
+// checkCircularReferences ensures there are no builds that can trigger themselves
+// due to an imagechangetrigger that matches the output destination of the image.
+func (c *AppConfig) checkCircularReferences(objects app.Objects) error {
+	for _, obj := range objects {
+		if bc, ok := obj.(*buildapi.BuildConfig); ok {
+			input := buildutil.GetImageStreamForStrategy(bc.Spec.Strategy)
+			if bc.Spec.Output.To != nil && input != nil &&
+				reflect.DeepEqual(input, bc.Spec.Output.To) {
```

Comment:
All we care about is whether this build can be triggered by an update to the same ImageStreamTag that it is outputting to.

So if either the output or the input are not ImageStreamTags, it's moot (you can't even have an ICT if your input is not an ImageStreamTag)